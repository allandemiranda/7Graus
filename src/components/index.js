export { default as Page } from './Page';
export { default as ScrollReset } from './ScrollReset';
export { default as StatusBullet } from './StatusBullet';