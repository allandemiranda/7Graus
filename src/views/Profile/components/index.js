export { default as Info } from './Info';
export { default as Header } from './Header';
export { default as Login } from './Login';
export { default as Location } from './Location';