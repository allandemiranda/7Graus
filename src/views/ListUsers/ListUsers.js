import React, { useState, useEffect } from 'react';
import { makeStyles } from '@material-ui/styles';
import { Typography } from '@material-ui/core';
import PropTypes from 'prop-types';
import CircularProgress from '@material-ui/core/CircularProgress';
import axios from 'utils/axios';
import { Page } from 'components';
import { Header, UserCard } from './components';

const useStyles = makeStyles(theme => ({
  root: {
    padding: theme.spacing(3)
  },
  results: {
    marginTop: theme.spacing(3)
  },
  paginate: {
    marginTop: theme.spacing(3),
    display: 'flex',
    justifyContent: 'center'
  }
}));

const ListUsers = (props) => {
  const classes = useStyles();
  const { match } = props;
  const { number } = match.params;
  const [users, setUsers] = useState([]);
  const [progress, setProgress] = useState(true);

  useEffect(() => {
    let mounted = true;

    const fetchProjects = () => {
      if(number){
        setTimeout(() => axios.get('/api/?results=' + number).then(response => {
          if (mounted) {
            setUsers(response.data.results);
          }
        }),
        2000);
      } else {
        setTimeout(() => axios.get('/api/?results=10').then(response => {
          if (mounted) {
            setUsers(response.data.results);
          }
        }),
        2000);
      }      
    };

    fetchProjects();

    return () => {
      mounted = false;
    };
  }, []);

  useEffect(() => {
    if (users.length != 0) {
      setProgress(false);
    }
  }, [users]);

  return (
    <Page 
      className={classes.root}
      title="User List"
    >
      <Header />
      {progress ? (
        <CircularProgress />
      ) : (
        <div className={classes.results}>
          <Typography
            color="textSecondary" 
            gutterBottom 
            variant="body2"
          >
            {users.length} Users found.
          </Typography>
          {users.map(user => (
            <UserCard
              user={user}
            />
          ))}
        </div>
      )}
    </Page>
  );
};

ListUsers.propTypes = {
  match: PropTypes.object.isRequired
};

export default ListUsers;
