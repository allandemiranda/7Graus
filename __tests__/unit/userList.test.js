/* eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import { Header } from '../../src/views/ListUsers/components';

describe('User List Tests', () => {
    it('Renders component correctly', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
    });

    it('Reload List', () => {
        delete window.location;
        window.location = { reload: jest.fn() };

        const tree = renderer.create(<Header />);
        const button = tree.root.findByProps({dataTestid:"reload-button"});
        button.props.onClick();  
              
        expect(button).toBeTruthy();  
    });
});