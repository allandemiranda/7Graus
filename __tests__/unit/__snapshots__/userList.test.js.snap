// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User List Tests Renders component correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-align-items-xs-flex-end MuiGrid-justify-xs-space-between"
  >
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <h2
        className="MuiTypography-root MuiTypography-overline MuiTypography-gutterBottom"
      >
        Users
      </h2>
      <h1
        className="MuiTypography-root MuiTypography-h3"
      >
        List
      </h1>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        dataTestid="reload-button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Reload
        </span>
      </button>
    </div>
  </div>
</div>
`;
